repo_resource/common.py

```diff
@@ -56,6 +72,42 @@ def remove_private_key_from_agent():
     atexit.unregister(ssh_agent_setup._kill_agent)
 
 
+def is_sha1(s):
+    return re.match(SHA1_PATTERN, s)
+
+
+def multi_run_wrapper(args):
+    return getRevision(*args)
+
+
+def getRevision(remote, remoteUrl, project, branch):
+    """
+    Get latest commit sha1 for revision
+    with git ls-remote command
+    without downloading the whole repo
+    """
+    # v1.0^{} is the commit referring to tag v1.0
+    # git ls-remote returns the tag sha1 if left as is
+    if branch.startswith('refs/tags'):
+        branch += '^{}'
+    try:
+        with redirect_stdout(sys.stderr):
+            # return tuple (remote/project, revision)
+            print('Fetching revision for '+remote+'/'+project+'...')
+            if is_sha1(branch):
+                return (remote + '/' + project, branch)
+            g = git.cmd.Git()
+            url, revision = (
+                remote + '/' + project,
+                g.ls_remote(remoteUrl+'/'+project, branch).split()[0]
+            )
+            print(url+': '+revision)
+            return (url, revision)
+    except Exception as e:
+        print('Cannot fetch project ', remoteUrl+'/'+project)
```

Comment:
This won't be shown in stderr so concourse will not display it

What's the purpose of these error messages in that case?

Should we print this to stderr instead?

Reply:
Printed in stderr now